.eslintrc.js

```diff
@@ -1,25 +1,27 @@
-const path = require('path');
-
+const { readFileSync } = require('node:fs');
+const path = require('node:path');
+const ts = require('typescript');
 const { version: reactVersion } = require('react/package.json');
 
+const tsconfigPath = ts.findConfigFile('./', ts.sys.fileExists);
+const { config } = ts.readConfigFile(tsconfigPath, ts.sys.readFile);
+const tsconfig = ts.parseJsonConfigFileContent(config, ts.sys, './');
```

Comment:
This gives us a fully parsed tsconfig, with `extends` rules applied and all `files` walked and included. Useful!